core/src/main/java/org/jdbi/v3/core/ExtensionRegistry.java

```diff
@@ -55,9 +55,10 @@ static ExtensionRegistry copyOf(ExtensionRegistry registry) {
         this.factories = new CopyOnWriteArrayList<>();
     }
 
+    @SuppressWarnings({"unchecked", "rawtypes"}) // working around https://bugs.eclipse.org/bugs/show_bug.cgi?id=499197
     ExtensionRegistry(List<Entry<? extends ExtensionConfig<?>>> factories) {
-        this.factories = factories.stream()
-                .map(Entry::copyOf)
+        this.factories = (List) factories.stream()
+                .map(e -> Entry.copyOf((Entry)e))
```

Comment:
Or rather, I think I generified Entry but only made `factories` a `List<Entry>` without the nested parameter.

I'm not really picky though, it's an internal class, so anything that does the job is fine.